docs/filesystems.txt

```diff
@@ -0,0 +1,144 @@
+.. _laravel-queues:
+
+======
+Queues
+======
+
+.. facet::
+   :name: genre
+   :values: tutorial
+
+.. meta::
+   :keywords: php framework, gridfs, code example
+
+Overview
+--------
+
+Using MongoDB to store files can be done using the
+`GridFS Adapter for Flysystem <https://flysystem.thephpleague.com/docs/adapter/gridfs/>`__.
+GridFS lets you store files of unlimited size in the same database as your data.
+This ensures the integrity of transactions and backups.
+
+
+Configuration
+-------------
+
+Before using the GridFS driver, you will need to install the Flysystem GridFS package via the Composer package manager:
+
+.. code-block:: bash
+
+   composer require league/flysystem-gridfs
+
+Configure `Laravel File Storage <https://laravel.com/docs/{+laravel-docs-version+}/filesystem>`__,
+to use the ``gridfs`` driver in ``config/filesystems.php``:
+
+.. code-block:: php
+
+   'disks' => [
+       'gridfs' => [
+           'driver' => 'gridfs',
+           'connection' => 'mongodb',
+           'database' => 'files',
+           'bucket' => 'fs',
+           'prefix' => '',
+           'read-only' => false,
+           'throw' => false,
```

Comment:
Is this the complete list of supported options, with default values repeated just so users know what they can set? Since you have a table below with all options, I'd consider reducing this code example to just the necessary options.

Reply:
> I find the complete example more synthetic and quicker to read

Same. If this example was annotated with comments to call out default values (and thus assist user's with reducing their own config) I think it'd stand well on its own and we wouldn't even need the table that follows.

Reply:
Added optional settings in comments. Keeping the table for details.